hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceImpl.java

```diff
@@ -132,9 +132,10 @@ protected HazelcastInstanceImpl(String name, Config config, NodeContext nodeCont
 
         try {
             logger = node.getLogger(getClass().getName());
-            lifecycleService.fireLifecycleEvent(STARTING);
```

Comment:
Then, we should have another rule for `HazelcastInstanceAware` interface to state that the system provides you the `HazelcastInstance` object after the start up is completed.